openpype/hosts/houdini/plugins/publish/validate_sop_output_node.py

```diff
@@ -2,12 +2,13 @@
 import pyblish.api
 from openpype.pipeline import PublishValidationError
 from openpype.pipeline.publish import RepairAction
+from openpype.hosts.houdini.api.action import SelectInvalidAction
 
 import hou
 
 
-class SelectInvalidAction(RepairAction):
-    label = "Select Invalid ROP"
+class SelectROPAction(RepairAction):
+    label = "Select ROP"
     icon = "mdi.cursor-default-click"
```

Comment:
Should we maybe change the `RepairAction` class so it allows customizing the method's name? Or even allows customizing that the method is on the action itself? A `SelectROPAction` would technically work for almost all Houdini validations because basically ALL of them have the `instance_node` available so we might even want to expose the `SelectROPAction` directly in `openpype.hosts.houdini.api.action` with selecting logic of its own.

E.g. this could be in the `action` lib:

```python
class SelectROPAction(pyblish.api.Action):
    on = "failed"  # This action is only available on a failed plug-in
    label = "Select ROP"
    icon = "mdi.cursor-default-click"

    def process(self, context, plugin):

        errored_instances = get_errored_instances_from_context(context)

        # Apply pyblish.logic to get the instances for the plug-in
        instances = pyblish.api.instances_by_plugin(errored_instances, plugin)

        # Get the invalid nodes for the plug-ins
        self.log.info("Finding invalid nodes..")
        rop_nodes = list()
        for instance in instances:
            path = instance.data.get("instance_node")
            if not path:
                continue

            node = hou.node(path)
            if not node:
                continue

            rop_nodes.append(node)

        hou.clearAllSelected()
        if invalid:
            self.log.info("Selecting ROP nodes: {}".format(
                ", ".join(node.path() for node in rop_nodes)
            ))
            for node in rop_nodes:
                node.setSelected(True)
                node.setCurrent(True)
        else:
            self.log.info("No ROP nodes found.")
```
And would be usable for all instances?

Reply:
@BigRoy we can definitely store this piece of information in `houdini/api/lib.py` for future use in validation.

Reply:
That'd be in `houdini/api/action.py` by the way, but yes. That was my intention.

Reply:
> we might even want to expose the `SelectROPAction` directly in `openpype.hosts.houdini.api.action` with selecting logic of its own.

Sounds interesting! I going to give it a try. 

>  if there are two pointcache instances with the same invalidation then it'd select both. It's unfortunately how the Action system works, it triggers per plugin - not per instance.

I re-read it, then I checked the `RepairAction`  class which has similar code like `SelectInvalidAction` 
So, I modified the selection code in `repair` method to 
```python
rop_node.setSelected(True)
rop_node.setCurrent(True)
```

and I faced this weird thing! it's confusing 
even if there's one instance that errored, `repair` method still runs when clicking `Select ROP` Button
so it works with each validator _i.e. plugin_  separately for all instances ?  

![image](https://github.com/ynput/OpenPype/assets/20871534/76055fa4-72f9-4d6f-bd78-b84051106023)

Reply:
Haha, wow. I'm quite blown away that this hasn't been found before. It looks like the call to `get_errored_instances_from_context` just filters whether that particular instance had ANY errors, not necessarily whether it had errors related to that particular plugin.

Could you try it so that the other instances don't have any OTHER errors too. Then it likely works. The fix shouldn't be that hard too.

Reply:
I made a test, publish 3 `pointcaches`
1. `pointcacheMain` which has no errors 
2. `pointcacheMain2` which points to a `SopNode` but has no path attribute 
3. `pointcacheMain3` which points to an `ObjNode` 


No 1 is the only instance that didn't fail, but it wasn't published as others failed 
No 2 has failed for another reason other than `Validate Output Node`
No 3 is the only one that didn't pass `Validate Output Node`

however, when clicking `SelectROPAction` it selects the associated ROPs with all errored instances
whether they had errors related to the plugin.

![image](https://github.com/ynput/OpenPype/assets/20871534/4546553b-986f-43d7-a146-9f8891f62890)

Reply:
Here's a draft PR which MIGHT solve the issue: https://github.com/ynput/OpenPype/pull/5240

Reply:
I will check it ASAP. 
and I have moved the SelectRopAction to `openpype.hosts.houdini.api.action`